server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/MailboxFactory.scala

```diff
@@ -152,7 +163,60 @@ class MailboxFactory @Inject() (subscriptionManager: SubscriptionManager) {
             namespace = namespace,
             rights = rights,
             quotas = quotas,
-            isSubscribed = retrieveIsSubscribed)))
+            isSubscribed = isSubscribed)))
+    }
+  }
+
+  def create(id: MailboxId, mailboxSession: MailboxSession, quotaLoader: QuotaLoader): SMono[Either[Exception, Mailbox]] = {
```

Comment:
IMHO SMono[Either[Exception... is not really a good return type.
Because it could be simplified with SMono[Mailbox].  Here we don't know at which level we should handle the exception.
If the exception in your either have a precise business meaning it should be a custom validation type.

Reply:
if not the flatter the better :)

Reply:
I'm a bit confused here as we went that way for the get all mailboxes version, with the MailboxValidation returning an Either (with you and Matthieu I think insisting on the Either? not sure to remember correctly)

In the case of exceptions regarding get mailboxes by id, you have two types:
* `MailboxNotFoundException` should not raise an error, but add the mailboxId in the `notFound` list in the response
* other errors have to throw

But could you detail a bit more? I'm not sure to get the full point sorry

Reply:
What i wanted to say is the Mono already convey the idea of an operation which could fail with an Exception.
So if the left type of the Either is an Exception it does not make a lot of sense to keep this nesting. 

But if you want to keep the Either because later we would refactor the left side of it to a custom Validation type, it's ok for me then.

Reply:
Hmmm Ok so you say we could keep the Either for `MailboxValidation` but after depending of if it succeeds or fail, we could return just a `Mono[Mailbox]` on the create method or raise an error correct? Because it would be propagated above by the Mono anyway